src/pyggdrasil/analyze/_rhat.py

```diff
@@ -44,6 +44,10 @@ def rhats(chains: np.ndarray) -> np.ndarray:
 
     Note:
         - May return NaN if the chains are too short and all values are the same
+        - May raise Out of memory error if the chains are too long, 100000 samples
+          still works, 1000000 does not - storing all truncated chains in memory
+           is too much even at 24 GB RAM ?!
+          TODO: find a way to calculate R-hat for longer chains
```

Comment:
The rhat calculation may be the reason for the OOM issues. - Thinning should circumvent the issue now - Rhats should be calculated for one length after another without storing all possible lengths simultaneously.

Below a wild thought on memory -  why this may not work: I manually create lists of all four chains' truncated lengths of the 4 chains to calculate rhat. This requires `4*chainlength` memory with a 100000 iteration chain already taking up 4MB in text form, I estimate 8MB (i.e. average length)* 100000 iterations = 800000 MB wait approx 800 GB ?  - is there something systemically flawed in this thought?